### PR TITLE
cli: support exporting login to stdout

### PR DESCRIPTION
**Pull request by kyrofa**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

-----
And instead of simply printing the ini file, base64-encode it so it can be more easily handled. This is intended to fit the use-case of sticking the exported login straight into the environment of a CI system instead of needing to encrypt a file.

### CODE REVIEW

File: snapcraft/cli/store.py [hunk 1]
```diff
@@ -341,19 +341,34 @@ def export_login(login_file: str, snaps: str, channels: str, acls: str,
...
-    # mypy doesn't have the opener arg in its stub. Ignore its warning
-    with open(login_file, 'w', opener=private_open) as f:  # type: ignore
-        store.conf.save(config_fd=f)
+    # Support a login_file of '-', which indicates a desire to print to stdout
+    if login_file.strip() == '-':
+        echo.info("\nExported login starts on next line:")
+        store.conf.save(config_fd=sys.stdout, encode=True)
+        print()
+
+        login_location = 'and printed above'
```
Comment (kalikiana) [inline snapcraft/cli/store.py]: I'd find it clearer to echo "Login successfully exported and printed above." here than to split up the string like that. Or alternatively set `login_location` to the whole string and use it below.

File: snapcraft/cli/store.py [hunk 2]
```diff
@@ -341,19 +341,34 @@ def export_login(login_file: str, snaps: str, channels: str, acls: str,
     # This is sensitive-- it should only be accessible by the owner
     private_open = functools.partial(os.open, mode=0o600)
 
-    # mypy doesn't have the opener arg in its stub. Ignore its warning
-    with open(login_file, 'w', opener=private_open) as f:  # type: ignore
-        store.conf.save(config_fd=f)
+    # Support a login_file of '-', which indicates a desire to print to stdout
```
Comment (kalikiana) [inline snapcraft/cli/store.py]: Mind adding the `-` to the login command documentation?

File: snapcraft/cli/store.py (hunk 1 above)
Comment (kyrofa) [reply snapcraft/cli/store.py]: Sure, done.

File: snapcraft/cli/store.py (hunk 2 above)
Comment (kyrofa) [reply snapcraft/cli/store.py]: Done.

**Pull request merged**